app/helpers/image_conversion_helper.rb

```diff
@@ -21,20 +21,35 @@ def resize_to_dimensions_and_convert_to_jpg(source_image_path:, attachment_type:
       return nil
     end
 
-    begin
-      actual_jpeg_to_save = Wasm::Thumbnail::Rb.resize_and_pad(file_bytes: file_bytes,
-                                                               width: dimensions[0],
-                                                               height: dimensions[1],
-                                                               size: target_file_size)
-    rescue RuntimeError
-      raise t("banner.upload_too_big")
-    end
+    actual_jpeg_to_save = retry_quality(file_bytes: file_bytes,
+                                        width: dimensions[0],
+                                        height: dimensions[1],
+                                        size: target_file_size,
+                                        quality: 50)
+
     new_filename = filename + "_resized"
     temp_file = Tempfile.new([new_filename, ".jpg"], binmode: true)
     temp_file.write(actual_jpeg_to_save)
     temp_file.path
   end
 
+  def retry_quality(file_bytes:, width:, height:, size:, quality:)
```

Comment:
Would you move this away from recursive? This could use quite a bit of space because of recursion, as worst case it's 10*file_bytes

Reply:
Done